File: src/main/java/br/com/leonardoferreira/jirareport/domain/vo/DynamicChart.java
package br.com.leonardoferreira.jirareport.domain.vo;

import br.com.leonardoferreira.jirareport.domain.embedded.Chart;
import com.fasterxml.jackson.annotation.JsonIgnore;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.io.Serializable;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class DynamicChart implements Serializable {
    private static final long serialVersionUID = 5429746843879854333L;

    private String name;

    private Chart<String, Double> leadTime;

    private Chart<String, Long> throughput;

    @JsonIgnore
    public String getId() {
        return name == null ? null : name.toLowerCase().replaceAll(" ", "");
    }

}

File: src/main/kotlin/br/com/jiratorio/domain/dynamicfield/DynamicChart.kt
package br.com.jiratorio.domain.dynamicfield

import br.com.jiratorio.domain.entity.embedded.Chart
import java.io.Serializable

data class DynamicChart(

    val name: String,

    val leadTime: Chart<String, Double>,

    val throughput: Chart<String, Int>

) : Serializable {
    companion object {
        private const val serialVersionUID = 5429746843879854333L
    }

}
